corehq/apps/userreports/tests/test_view.py

```diff
@@ -35,32 +39,9 @@ class ConfigurableReportTestMixin(object):
     case_type = "CASE_TYPE"
 
     @classmethod
-    def _new_case(cls, properties):
-        id = uuid.uuid4().hex
-        case_block = CaseBlock(
-            create=True,
-            case_id=id,
-            case_type=cls.case_type,
-            update=properties,
-        ).as_text()
-        with drop_connected_signals(sql_case_post_save):
-            submit_case_blocks(case_block, domain=cls.domain)
-        return CommCareCase.objects.get_case(id, cls.domain)
-
-    @classmethod
-    def _delete_everything(cls):
-        delete_all_cases()
-        for config in DataSourceConfiguration.all():
-            config.delete()
-        delete_all_report_configs()
-
-
-class ConfigurableReportViewTest(ConfigurableReportTestMixin, TestCase):
-
-    def _build_report_and_view(self, request=HttpRequest()):
-        # Create report
-        data_source_config = DataSourceConfiguration(
-            domain=self.domain,
+    def _sample_data_source_config(cls):
+        return DataSourceConfiguration(
+            domain=cls.domain,
```

Comment:
I simply moved the data source initialization to the base class in a class method ` _sample_data_source_config ` for reuse. The diff has become tricky.